relayer/provider/provider.go

```diff
@@ -101,6 +102,7 @@ type ChainProvider interface {
 // Do we need intermediate types? i.e. can we use the SDK types for both substrate and cosmos?
 type QueryProvider interface {
 	// chain
+	Block(ctx context.Context, height *int64) (*ctypes.ResultBlock, error)
```

Comment:
The `Provider` interfaces have to be generic enough to work with non-Cosmos/Tendermint chains so we can't have a return type imported from tendermint.

What we probably want to do instead is add a function like `BlockTime() (int64, error)` to `ChainProvider`  for retrieving the unix timestamp since that's the only thing we need from the `ResultBlock`.

https://github.com/cosmos/relayer/blob/457f2ed485463535ecd66369c45a22e957c99200/relayer/provider/cosmos/provider.go#L1141

I think that would be flexible enough to work for chains from other ecosystems.